addons/purchase/models/purchase_order_line.py

```diff
@@ -369,54 +370,10 @@ def _compute_price_unit_and_date_planned_and_name(self):
-    def _compute_product_packaging_id(self):
-        for line in self:
-            # remove packaging if not match the product
-            if line.product_packaging_id.product_id != line.product_id:
-                line.product_packaging_id = False
-            # suggest biggest suitable packaging matching the PO's company
-            if line.product_id and line.product_qty and line.product_uom_id:
-                suggested_packaging = line.product_id.packaging_ids\
-                        .filtered(lambda p: p.purchase and (p.product_id.company_id <= p.company_id <= line.company_id))\
-                        ._find_suitable_product_packaging(line.product_qty, line.product_uom_id)
-                line.product_packaging_id = suggested_packaging or line.product_packaging_id
-
-    @api.onchange('product_packaging_id')
-    def _onchange_product_packaging_id(self):
-        if self.product_packaging_id and self.product_qty:
-            newqty = self.product_packaging_id._check_qty(self.product_qty, self.product_uom_id, "UP")
-            if float_compare(newqty, self.product_qty, precision_rounding=self.product_uom_id.rounding) != 0:
-                return {
-                    'warning': {
-                        'title': _('Warning'),
-                        'message': _(
-                            "This product is packaged by %(pack_size).2f %(pack_name)s. You should purchase %(quantity).2f %(unit)s.",
-                            pack_size=self.product_packaging_id.qty,
-                            pack_name=self.product_id.uom_id.name,
-                            quantity=newqty,
-                            unit=self.product_uom_id.name
-                        ),
-                    },
-                }
-
-    @api.depends('product_packaging_id', 'product_uom_id', 'product_qty')
-    def _compute_product_packaging_qty(self):
-        self.product_packaging_qty = 0
-        for line in self:
-            if not line.product_packaging_id:
-                continue
-            line.product_packaging_qty = line.product_packaging_id._compute_qty(line.product_qty, line.product_uom_id)
-
-    @api.depends('product_packaging_qty')
-    def _compute_product_qty(self):
+    @api.depends('product_id')
+    def _compute_product_uom_id(self):
         for line in self:
-            if line.product_packaging_id:
-                packaging_uom = line.product_packaging_id.product_uom_id
-                qty_per_packaging = line.product_packaging_id.qty
-                product_qty = packaging_uom._compute_quantity(line.product_packaging_qty * qty_per_packaging, line.product_uom_id)
-                if float_compare(product_qty, line.product_qty, precision_rounding=line.product_uom_id.rounding) != 0:
-                    line.product_qty = product_qty
+            line.product_uom_id = line.product_id.seller_ids.filtered(lambda s: s.partner_id == line.partner_id).product_uom_id or line.product_id.uom_id if line.product_id else False
```

Comment:
Same issue as in `account_move_line`, there may be multiple `seller_ids` with the same `partner_id` with potentially different UoMs as well. This also overlaps with `_product_id_change()`. We still need to convert the onchanges in purchase to computes, so this is a bit of a mess for how to fix. Do you actually need this compute for now? It seems like this method isn't used anywhere